### PR TITLE
Yaml predeployment

### PR DESCRIPTION
**Pull request by mehros**
1- add json and yaml types as supported type 
2 -add json and yaml scenrio in realm/valiadtion
3- raise error message if prancer can not find opa binary 
4- raise error message if prancer has any problem to run opa binary

### CODE REVIEW

File: config.ini [hunk 1]
```diff
@@ -15,7 +15,7 @@ database = NONE
 
 [OPA]
 opa =  true
-opaexe = $OPA_HOME/opa
+opaexe = /usr/bin/opa
```
Comment (farchide) [inline config.ini]: Why this is changed?

File: realm/test-gitConnector.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (farchide) [inline realm/test-gitConnector.json]: do not put test json files in this repo
remove the file

File: realm/validation/scenario-json/output-test.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-json/output-test.json]: remove this file

File: realm/validation/scenario-json/samp.rego [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package rule
```
Comment (farchide) [inline realm/validation/scenario-json/samp.rego]: remove this

File: realm/validation/scenario-json/snapshot.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-json/snapshot.json]: remove this file

File: realm/validation/scenario-json/snapshot_gen.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-json/snapshot_gen.json]: remove this file

File: realm/validation/scenario-json/snapshots/YAML_TEMPLATE_1 [hunk 7]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-json/snapshots/YAML_TEMPLATE_1]: remove this file

File: realm/validation/scenario-json/test.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-json/test.json]: remove this file

File: realm/validation/scenario-yaml/samp.rego [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package rule
```
Comment (farchide) [inline realm/validation/scenario-yaml/samp.rego]: remove this file

File: realm/validation/scenario-yaml/snapshot.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-yaml/snapshot.json]: remove this file

File: realm/validation/scenario-yaml/test.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (farchide) [inline realm/validation/scenario-yaml/test.json]: remove this file

**Review by farchide — CHANGES_REQUESTED** [review]

**Review by farchide — APPROVED** [review]

File: src/processor/comparison/interpreter.py [hunk 12]
```diff
@@ -6,6 +6,7 @@
 import json
 import os
 import re
+import sys
```
Comment (vatsalgit5118) [inline src/processor/comparison/interpreter.py]: If `sys` import is not used anywhere then remove it.

File: src/processor/template_processor/base/base_template_processor.py [hunk 13]
```diff
@@ -284,6 +284,10 @@ def populate_all_template_snapshot(self):
             root_dir_path = self.repopath 
 
         self.paths = get_field_value(self.node, 'paths')
+        self.type = get_field_value(self.node, 'type')
+        
+        if self.type in ["yaml","json"] and self.paths  is None:
```
Comment (vatsalgit5118) [inline src/processor/template_processor/base/base_template_processor.py]: The validation is already there for check `self.paths` is exist or not.
Just update the error message in Line no: 302

File: config.ini (hunk 1 above)
Comment (devtrap) [reply config.ini]: it was changed to run in my local
returned to the previous state

File: realm/test-gitConnector.json (hunk 2 above)
Comment (devtrap) [reply realm/test-gitConnector.json]: extra json file removed

File: realm/validation/scenario-json/output-test.json (hunk 3 above)
Comment (devtrap) [reply realm/validation/scenario-json/output-test.json]: file removed

File: realm/validation/scenario-json/samp.rego (hunk 4 above)
Comment (devtrap) [reply realm/validation/scenario-json/samp.rego]: removed

File: realm/validation/scenario-json/snapshot.json (hunk 5 above)
Comment (devtrap) [reply realm/validation/scenario-json/snapshot.json]: file removed

File: realm/validation/scenario-json/test.json (hunk 8 above)
Comment (devtrap) [reply realm/validation/scenario-json/test.json]: file removed

File: realm/validation/scenario-yaml/samp.rego (hunk 9 above)
Comment (devtrap) [reply realm/validation/scenario-yaml/samp.rego]: file removed

File: realm/validation/scenario-yaml/snapshot.json (hunk 10 above)
Comment (devtrap) [reply realm/validation/scenario-yaml/snapshot.json]: file removed

File: realm/validation/scenario-yaml/test.json (hunk 11 above)
Comment (devtrap) [reply realm/validation/scenario-yaml/test.json]: file removed

File: src/processor/comparison/interpreter.py (hunk 12 above)
Comment (devtrap) [reply src/processor/comparison/interpreter.py]: extra package removed

File: src/processor/template_processor/base/base_template_processor.py (hunk 13 above)
Comment (devtrap) [reply src/processor/template_processor/base/base_template_processor.py]: condition removed and the type will check in another func

**Review by vatsalgit5118 — APPROVED** [review]

**Pull request merged**